packages/iov-bns-governance/src/governor.ts

```diff
@@ -0,0 +1,100 @@
+import BN = require("bn.js");
+
+import { Address, Identity, WithCreator } from "@iov/bcp";
+import {
+  bnsCodec,
+  BnsConnection,
+  BnsTx,
+  CreateProposalTx,
+  ElectionRule,
+  Electorate,
+  Proposal,
+  TallyTx,
+  VoteOption,
+  VoteTx,
+} from "@iov/bns";
+
+import { ProposalOptions, ProposalType } from "./proposals";
+import { groupByCallback, maxWithComparatorCallback } from "./utils";
+
+export interface GovernorOptions {
+  readonly connection: BnsConnection;
+  readonly identity: Identity;
+}
+
+export class Governor {
+  private readonly connection: BnsConnection;
+  private readonly identity: Identity;
+  private readonly address: Address;
+
+  public constructor({ connection, identity }: GovernorOptions) {
+    this.connection = connection;
+    this.identity = identity;
+    this.address = bnsCodec.identityToAddress(this.identity);
+  }
+
+  public async getElectorates(): Promise<readonly Electorate[]> {
+    const electorates = await this.connection.getElectorates();
+    return electorates.filter(({ electors }) => Object.keys(electors).some(key => key === this.address));
+  }
+
+  public async getElectionRules(electorateId: number): Promise<readonly ElectionRule[]> {
+    const electionRules = await this.connection.getElectionRules();
+    const filteredRules = electionRules.filter(rule => rule.electorateId === electorateId);
+
+    if (filteredRules.length === 0) {
+      throw new Error("No election rule found for electorate");
+    }
+
+    const groupedRules = groupByCallback(filteredRules, rule => rule.id);
+    return groupedRules.map(group =>
+      maxWithComparatorCallback(group.values, (rule1, rule2) => rule1.version - rule2.version),
+    );
+  }
+
+  public async getProposals(): Promise<readonly Proposal[]> {
+    const electorates = await this.getElectorates();
+    const proposals = await this.connection.getProposals();
+    return proposals.filter(({ electorate }) => {
+      const electorateId = new BN(electorate.id).toNumber();
+      return electorates.some(({ id }) => id === electorateId);
+    });
+  }
+
+  public async createProposalTx(options: ProposalOptions): Promise<BnsTx & WithCreator> {
```

Comment:
* Return type is always `CreateProposalTx & WithCreator`, just with different `action` properties
* Since this creates a `CreateProposalTx`, the name should be `createCreateProposalTx`. Alternatively, we can prefix all 3 methods with `make` or `build`

Reply:
I like `build`.

Reply:
[I like turtles](https://www.youtube.com/watch?v=CMNry4PE93Y).